src/Parser/Parser.hs

```diff
@@ -379,10 +379,11 @@ valdef :: Parser (ValDef SourcePos)
 valdef = do
   (Sig n t) <- sig
   b <- getCtype
-  case b of
-    Just (Plain (X Board set))
-      | S.null set -> (BVal (Sig n t)) <$> many1 (boardeqn n) <*> getPosition
-    _ -> (Val (Sig n t)) <$> (equation) <*> getPosition
+  let val = (Val (Sig n t)) <$> (equation) <*> getPosition in
+     case b of
+        Just (Plain (X Board set))
+           | S.null set -> ((BVal (Sig n t)) <$> many1 (boardeqn n) <*> getPosition) <|> val
+        _ -> val
```

Comment:
Was there a particular reason for this change?

Reply:
Yes. This is the fix for #75. If a signature has type `Board`, it is not necessarily a board equation. If parsing as a board equation fails, it attempts to parse as a regular equation. 

That is for a case like:

```
b2 : Board
b2 = b1
```